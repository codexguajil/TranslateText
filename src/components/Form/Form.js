import React, { Component } from 'react';

export default class Form extends Component {
  constructor() {
    super();
    this.state = {
      text: '',
    }
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit = (e) => {
    e.preventDefault()
    this.props.handleSubmit(this.state.text)
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input value={this.state.text}
               name="text"
               onChange={this.handleChange}
        />
        <button>Translate</button>
      </form>
    )
  }
}