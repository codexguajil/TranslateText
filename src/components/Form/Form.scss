.form {
  background: none;
  display: inline-flex;
  flex-direction: column;
  padding-top: 26px;
  padding-bottom: 26px;
  width: 100%;
}

.header {
  margin-top: 200px;
}

.input {
  border-radius: 15px;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
  font-size: 26px;
  height: 127px;
  letter-spacing: 1.2px;
  padding: 10px;
  padding-bottom: 35px;
  word-break: break-word;
  width: 977px;
  &:hover {
    background: hsla(60,100%,50%, .1);
    cursor: auto;
  }
  &:focus {
    background: hsla(60,100%,50%, .1);
    outline: none;
  }
}

#input-cont {
  display: inline-flex;
  flex-direction: column;
  margin: auto;
  width: 977px;
}

.translate-btn {
  background: #FFF;
  border-bottom-left-radius: 20px;
  border-bottom-right-radius: 20px;
  color: black;
  font-size: 20px;
  height: 40px;
  margin: auto;
  margin-top: -5px;
  width: 1000px;
  &:hover {
    background: hsla(300,100%,30%, .1);
    border-bottom: 1px solid black;
    border-left: 1px solid black;
    cursor: pointer;
    font-weight: bolder;
    letter-spacing: 1.3px;
    text-shadow: 1px 1px lavender;
    transition: background-color .5s ease;
  }
  &:focus {
    outline: none;
  }
}

.nav-link {
  border: none;
  color: black;
  font-size: 26px;
  position: absolute;
  right: 25px;
  top: 25px;
  width: 300px;
  &:hover {
    color: hsla(300,100%,30%, .4);
    cursor: pointer;
    text-decoration: underline;
  }
  &:focus {
    outline: none;
  }
}