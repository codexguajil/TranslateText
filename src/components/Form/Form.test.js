import React from 'react'
import {shallow} from 'enzyme'
import {Form} from './Form'

describe('Form', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Form />
    )
  })
  it('should match the snapshot', () => {
    let wrapper = shallow(
      <Form />
    )
    
    expect(wrapper).toMatchSnapshot()
  })

  it('should update state on change', () => {
    wrapper.setState({text: ''})
    expect(wrapper.state('text')).toEqual('')

    let e = {
      target: {
        name: 'text',
        value: 'hello'
      }
    }

    wrapper.instance().handleChange(e)

    expect(wrapper.state('text')).toEqual('hello')
  })

  it('should call handleSubmit on click translate button', () => {
    let props = {
      handleError: jest.fn()
    }

    let wrapper = shallow(
      <Form {...props} />
      )

    wrapper.instance().handleSubmit = jest.fn()
    let mockEvent = { preventDefault: jest.fn()}
    wrapper.find('.form').simulate('submit', mockEvent)
    expect(mockEvent.preventDefault).toHaveBeenCalled()
  })

  it('should reroute to translations if See Translations is clicked', () => {
    let props = {
      handleSubmit: jest.fn(),
      history: { push: jest.fn() }
    }

    let wrapper = shallow(
      <Form {...props} />
      )

      wrapper.setState({text: 'hello'})
      let mockEvent = { preventDefault: jest.fn()}
      wrapper.instance().handleSubmit(mockEvent)
      expect(wrapper.instance().props.handleSubmit).toHaveBeenCalled()
  })

  it('should return an error if Translate is clicked without any text in state', () => {
    let props = {
      handleError: jest.fn(),
      handleSubmit: jest.fn(),
      history: { push: jest.fn() }
    }

    let wrapper = shallow(
      <Form {...props} />
      )

      let mockEvent = { preventDefault: jest.fn()}
      wrapper.instance().handleSubmit(mockEvent)
      expect(wrapper.instance().props.handleError).toHaveBeenCalled()
  })

  it('should have props', () => {
    let props = {
      translations: ['bonjour'],
      handleError: jest.fn(),
      handleSubmit: jest.fn(),
      history: { push: jest.fn() }
    }

    let wrapper = shallow(
      <Form {...props} />
    )

    wrapper.instance().handleClick()
    expect(wrapper.instance().props).toEqual(props)
  })

  it('should return an error if See Translations is clicked without any translations in store', () => {
    let props = {
      translations: [],
      handleError: jest.fn(),
      handleSubmit: jest.fn(),
      history: { push: jest.fn() }
    }

    let wrapper = shallow(
      <Form {...props} />
      )

      let mockEvent = { preventDefault: jest.fn()}
      wrapper.instance().handleClick(mockEvent)
      expect(wrapper.instance().props.handleError).toHaveBeenCalled()
  })
})