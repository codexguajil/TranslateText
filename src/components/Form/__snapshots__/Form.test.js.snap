// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="nav-link"
          onClick={[Function]}
        >
          See Translations.
        </button>,
        <Link
          className="header"
          to="/"
        >
          Translate to French.
        </Link>,
        <form
          className="form"
          onSubmit={[Function]}
        >
          <div
            id="input-cont"
          >
            <textarea
              className="input"
              name="text"
              onChange={[Function]}
              placeholder="Type your english text to translate here! Expand the text area for more space."
              value=""
            />
            <button
              className="translate-btn"
            >
              Translate
            </button>
          </div>
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "See Translations.",
          "className": "nav-link",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "See Translations.",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Translate to French.",
          "className": "header",
          "to": "/",
        },
        "ref": null,
        "rendered": "Translate to French.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            id="input-cont"
          >
            <textarea
              className="input"
              name="text"
              onChange={[Function]}
              placeholder="Type your english text to translate here! Expand the text area for more space."
              value=""
            />
            <button
              className="translate-btn"
            >
              Translate
            </button>
          </div>,
          "className": "form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <textarea
                className="input"
                name="text"
                onChange={[Function]}
                placeholder="Type your english text to translate here! Expand the text area for more space."
                value=""
              />,
              <button
                className="translate-btn"
              >
                Translate
              </button>,
            ],
            "id": "input-cont",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input",
                "name": "text",
                "onChange": [Function],
                "placeholder": "Type your english text to translate here! Expand the text area for more space.",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Translate",
                "className": "translate-btn",
              },
              "ref": null,
              "rendered": "Translate",
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="nav-link"
            onClick={[Function]}
          >
            See Translations.
          </button>,
          <Link
            className="header"
            to="/"
          >
            Translate to French.
          </Link>,
          <form
            className="form"
            onSubmit={[Function]}
          >
            <div
              id="input-cont"
            >
              <textarea
                className="input"
                name="text"
                onChange={[Function]}
                placeholder="Type your english text to translate here! Expand the text area for more space."
                value=""
              />
              <button
                className="translate-btn"
              >
                Translate
              </button>
            </div>
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "See Translations.",
            "className": "nav-link",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "See Translations.",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Translate to French.",
            "className": "header",
            "to": "/",
          },
          "ref": null,
          "rendered": "Translate to French.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              id="input-cont"
            >
              <textarea
                className="input"
                name="text"
                onChange={[Function]}
                placeholder="Type your english text to translate here! Expand the text area for more space."
                value=""
              />
              <button
                className="translate-btn"
              >
                Translate
              </button>
            </div>,
            "className": "form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <textarea
                  className="input"
                  name="text"
                  onChange={[Function]}
                  placeholder="Type your english text to translate here! Expand the text area for more space."
                  value=""
                />,
                <button
                  className="translate-btn"
                >
                  Translate
                </button>,
              ],
              "id": "input-cont",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input",
                  "name": "text",
                  "onChange": [Function],
                  "placeholder": "Type your english text to translate here! Expand the text area for more space.",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Translate",
                  "className": "translate-btn",
                },
                "ref": null,
                "rendered": "Translate",
                "type": "button",
              },
            ],
            "type": "div",
          },
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
